pkg/typesystem/resolver_test.go

```diff
@@ -0,0 +1,50 @@
+package typesystem
+
+import (
+	"context"
+	"testing"
+
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/golang/mock/gomock"
+	"github.com/oklog/ulid/v2"
+	mockstorage "github.com/openfga/openfga/pkg/storage/mocks"
+	"github.com/stretchr/testify/require"
+	openfgav1 "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+func TestMemoizedTypesystemResolverFunc(t *testing.T) {
```

Comment:
todo(@jon-whit) - You should also add one more test for the case where the authorization model identifier is `""`.